hydra-node/src/Hydra/Chain/Blockfrost/Client.hs

```diff
@@ -63,153 +83,83 @@ runBlockfrostM prj action = do
-  Blockfrost.PlutusV3 -> Just PlutusV3
-  Blockfrost.Timelock -> Nothing
-
-toCardanoPoolId :: Blockfrost.PoolId -> Hash StakePoolKey
-toCardanoPoolId (Blockfrost.PoolId textPoolId) =
-  case deserialiseFromRawBytesHex (encodeUtf8 textPoolId) of
-    Left err -> error (show err)
-    Right pool -> pool
-
-toCardanoUTxO :: [Blockfrost.AddressUtxo] -> AddressInEra -> UTxO' (TxOut CtxUTxO)
-toCardanoUTxO utxos addr = UTxO.fromList (toEntry <$> utxos)
- where
-  toEntry :: Blockfrost.AddressUtxo -> (TxIn, TxOut CtxUTxO)
-  toEntry utxo = (toCardanoTxIn utxo, toCardanoTxOut utxo addr)
-
-toCardanoTxIn :: Blockfrost.AddressUtxo -> TxIn
-toCardanoTxIn Blockfrost.AddressUtxo{_addressUtxoTxHash = Blockfrost.TxHash{unTxHash}, _addressUtxoOutputIndex} =
-  case deserialiseFromRawBytesHex (encodeUtf8 unTxHash) of
-    Left err -> error (show err)
-    Right txid -> TxIn txid (TxIx (fromIntegral _addressUtxoOutputIndex))
-
--- REVIEW! TxOutDatumNone and ReferenceScriptNone
-toCardanoTxOut :: Blockfrost.AddressUtxo -> AddressInEra -> TxOut CtxUTxO
-toCardanoTxOut addrUTxO addr =
-  TxOut addr (toCardanoValue _addressUtxoAmount) TxOutDatumNone ReferenceScriptNone
- where
-  Blockfrost.AddressUtxo{_addressUtxoAmount, _addressUtxoDataHash, _addressUtxoInlineDatum, _addressUtxoReferenceScriptHash} = addrUTxO
-
-toCardanoPolicyId :: Text -> PolicyId
-toCardanoPolicyId pid =
-  case deserialiseFromRawBytesHex (encodeUtf8 pid) of
-    Left err -> error (show err)
-    Right p -> p
+  BlockfrostClientT IO [TxId]
+publishHydraScripts sk = do
+  Blockfrost.Genesis
+    { _genesisNetworkMagic = networkMagic
+    , _genesisSystemStart = systemStart'
+    } <-
+    queryGenesisParameters
+  pparams <- queryProtocolParameters
+  let address = Blockfrost.Address (vkAddress networkMagic)
+  let networkId = toCardanoNetworkId networkMagic
+  stakePools' <- Blockfrost.listPools
+  let stakePools = Set.fromList (toCardanoPoolId <$> stakePools')
+  let systemStart = SystemStart $ posixSecondsToUTCTime systemStart'
+  eraHistory <- queryEraHistory
+  -- addressUTxO <- Blockfrost.getAddressUtxos address
+  addressUTxO <- Blockfrost.getAddressUtxos address
 
-toCardanoAssetName :: Text -> AssetName
-toCardanoAssetName = AssetName . encodeUtf8
+  cardanoUTxO <-
+    foldMapM
+      ( \Blockfrost.AddressUtxo
+           { Blockfrost._addressUtxoAddress
+           , Blockfrost._addressUtxoTxHash = Blockfrost.TxHash{unTxHash}
+           , Blockfrost._addressUtxoOutputIndex
+           , Blockfrost._addressUtxoAmount
+           , Blockfrost._addressUtxoBlock
+           , Blockfrost._addressUtxoDataHash
+           , Blockfrost._addressUtxoInlineDatum
+           , Blockfrost._addressUtxoReferenceScriptHash
+           } ->
+            let txin = toCardanoTxIn unTxHash _addressUtxoOutputIndex
+             in toCardanoUTxO networkId txin _addressUtxoAddress _addressUtxoReferenceScriptHash _addressUtxoDataHash _addressUtxoAmount _addressUtxoInlineDatum
+      )
+      addressUTxO
```

Comment:
Isn't this the same as `queryUTxO`?

Reply:
Apparently it is :) I used queryUTxO directly, thanks